src/hotspot/share/prims/jvm.cpp

```diff
@@ -4044,3 +4044,22 @@ JVM_ENTRY(void, JVM_VirtualThreadUnmountEnd(JNIEnv* env, jobject vthread, jboole
   fatal("Should only be called with JVMTI enabled");
 #endif
 JVM_END
+
+/*
+ * Return the current class's class file version.  The low order 16 bits of the
+ * the returned jint contains the class's major version.  The high order 16 bits
+ * contains the class's minor version.
+ */
+JVM_ENTRY(jint, JVM_GetClassFileVersion(JNIEnv* env, jclass current))
+  oop mirror = JNIHandles::resolve_non_null(current);
+  if (java_lang_Class::is_primitive(mirror)) {
+    // return latest major version and minor version of 0.
+    return JVM_CLASSFILE_MAJOR_VERSION;
+  }
+  assert(!java_lang_Class::as_Klass(mirror)->is_array_klass(), "unexpected array class");
+
+  Klass* c = java_lang_Class::as_Klass(mirror);
+  assert(c->is_instance_klass(), "must be");
+  InstanceKlass* ik = InstanceKlass::cast(c);
+  return (ik->minor_version() << 16) | ik->major_version();
```

Comment:
I'm curious why the format is minor:major rather than major:minor ?

Reply:
This was requested by @RogerRiggs 

Major high vs low bits requires either:
  High bits:   val >>>16
  Low bits:     val & 0xffff

Except for preview, the minor version is 0, so putting major version  in the low bits allows trivial comparison (without masking); but may lead to bugs due to carelessness.

Reply:
Okay makes sense.